lib/req_telemetry.ex

```diff
@@ -5,10 +5,10 @@ defmodule ReqTelemetry do
   require Logger
 
   @type options :: boolean() | [option]
-  @type option :: {:adapter, boolean()} | {:pipeline, boolean()}
+  @type option :: {:adapter, boolean()} | {:pipeline, boolean()} | {:metadata, map()}
 
-  @default_opts %{adapter: true, pipeline: true}
-  @no_emit_opts %{adapter: false, pipeline: false}
+  @default_opts %{adapter: true, pipeline: true, metadata: %{}}
+  @no_emit_opts %{adapter: false, pipeline: false, metadata: %{}}
```

Comment:
Can we change the default to `nil`?

I prefer `nil` here because it allows for easier pattern matching -- the presence of any map means that someone set metadata at some point. Contrived example, but:

```elixir
defp handle_metadata(%{metadata: %{} = metadata}), do: #...
defp handle_metadata(_), do: :ignore
```

Reply:
Done